tap_mailchimp/discover.py

```diff
@@ -1,22 +1,31 @@
 from singer.catalog import Catalog, CatalogEntry, Schema
-
-from tap_mailchimp.schema import get_schemas, PKS
+from tap_mailchimp.streams import STREAMS
 
 def discover():
-    schemas, field_metadata = get_schemas()
     catalog = Catalog([])
 
-    for stream_name, schema_dict in schemas.items():
-        schema = Schema.from_dict(schema_dict)
-        metadata = field_metadata[stream_name]
-        pk = PKS[stream_name]
+    for _, stream_obj in STREAMS.items():
+        schemas, field_metadata = stream_obj.get_schema()
+
+        # loop over the schema and prepare catalog
+        for stream_name, schema_dict in schemas.items():
```

Comment:
It's better to use get_schemas() which returns schema for all the streams. It will reduce one for the loop above. Ideally, add get_schemas() function to schema.py as other standard tap structure, and remove get_schema from the stream class.